### PR TITLE
docker: add basic Dockerfile

### PR DESCRIPTION
**Pull request by Matt-Hurd**
I like my multi-stage builds. Creates a very small image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.16-alpine3.13 as builder
+
+WORKDIR /app
+COPY . .
+RUN apk add --no-cache make ca-certificates
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o gql_server -ldflags="-w -s" ./app/graphql_server/main.go
+
+FROM scratch
```
Comment (akleemans) [inline Dockerfile:8]: Those are two images in one file, right? (I've never seen two `FROM` blocks in one file.)

**Comment by Matt-Hurd** [comment]
Fixed this. Originally I was doing a `COPY . .` which was causing extra docker cache invalidations. Much cleaner now.

File: README.md [hunk 2]
```diff
@@ -36,4 +36,9 @@ Running tests:
 - `make test`
 
 Running benchmarks:
-- `make bench`
\ No newline at end of file
+- `make bench`
+
+## Running in a container
+- `docker build . -t graphql_server`
+- `docker run --rm -p 3030:3030 -d graphql_server`
```
Comment (akleemans) [inline README.md:43]: I personally would not include `--rm` in the default `docker run` command but I guess that's a matter of preference.

**Review by akleemans — APPROVED** [review]
Image can be built and run, did some tests with Postman which worked fine.

**Comment by zdylag** [comment]
When you fix the conflicts can you also update the commit to `docker: ...`? I will try and address https://github.com/speedrun-website/leaderboard-backend/issues/32 soon.

**Pull request merged**